### ISSUE TITLE
Provide some API for dealing with leading whitespaces

### ISSUE THREAD

**Issue by gabro** [scalacenter/scalafix#370]
When removing all the tokens of a line it's very common to be willing to remove also the leading whitespaces including the newline.

e.g.

removing `bar` from

```
foo
  bar
baz
```

should result in

```
 foo
 baz
```

as opposed to

```
foo
  
baz
```

Currently doing this is fairly tedious. We should provide a better API for this use case.

For example

```scala
ctx.removeLeadingWhitespace(t, includingNewline = true)
```

or even a modifier on existing methods (binary compat not going to be happy, though):

```scala
ctx.removeTokens(t.tokens, removeLeadingWhitespaces = true)
```

**Comment by ShaneDelmore** [comment]
I’m bikeshedding here but given how common this is, what about shorter more common names like:
- trim
- trimLeft
- trimRight

**Comment by olafurpg** [comment]
Fixed in #474
